### DISCUSSION TITLE
In dev mode after hot reload TypeError: __webpack_require__.C is not a function error

### DISCUSSION THREAD

**Discussion by rnnyrk** [vercel/next.js#46145] [Help]
I'm trying out the new app directory. Tyring to build a small website, fetching data from my Notion using their API and displaying the results. I have two basic pages. This is the "parent" page in `app/resources/page.tsx`. 

```typescript
import { ResourcesTweets } from 'modules/resources';
import { Suspense } from 'react';

import LoadingSvg from 'vectors/loading.svg';

const Resources = () => {
  return (
    <section>
      <h2>Tweets</h2>
      <Suspense fallback={<LoadingSvg />}>
        {/* @ts-expect-error Server Component */}
        <ResourcesTweets />
      </Suspense>
    </section>
  );
};

export default Resources;
```

The `ResourcesTweets` components is living outside of my app directory (I do not know if this might be an issue?) in `components/modules/resources/ResourcesTweets.tsx`. 

```typescript
export const ResourcesTweets = async () => {
  let resources: null | Resource[] = null;

  try {
    const res = await fetch('http://localhost:3000/api/notion?type=twitter-saves', {
      headers: { 'Content-Type': 'application/json' },
      method: 'GET',
    });

    const data = await res.json();
    resources = data;
  } catch (error) {
    console.error(error);
  }

  return (
    <div className="flex">
      {resources &&
        resources.map((resource) => (
          <Card
            key={resource.id}
            title={resource.title}
            tags={resource.tags?.map((tag) => tag.name)}
          />
        ))}
    </div>
  );
};
```

Both are relatively simple and I expect both to be rendered without any issue as a server component. Although during development I keep running into this error when the app is hot reloading:

<img width="1114" alt="Screenshot 2023-02-20 at 09 07 34" src="https://user-images.githubusercontent.com/4519142/220071324-bd7aab21-5dcd-4c6e-9f96-d8bc8eeb2437.png">

It doesn't really tell me anything useful. Als googling the issue doesn't help out. Since I do not think it's a bug, but rather some implementation issue on my side, I thought this was the right place to ask for help.

**Comment by karlhorky** [comment]
@rnnyrk did you find a way to move past this? If so, what was your solution?

**Reply by karlhorky** [reply]
For our `__webpack_require__.n is not a function` error, it was related to having a `react@19.0.0-rc.0` + `react-dom@19.0.0-rc.0` version which is incompatible with latest canary `next@15.0.0-canary.67`:

- https://github.com/vercel/next.js/issues/67783

**Reply by rnnyrk** [reply]
@karlhorky Not quite sure what the solution was to be honest. Remember something like mismatch versions, but not 100% sure

**Reply by karlhorky** [reply]
Ok thanks for the answer!

After speaking with Sebastian Silbermann, I learned that apparently `react@19.0.0-rc.0` + `react-dom@19.0.0-rc.0` are the correct versions (although there are known problems in the webpack dev server, with [a PR open to work around them](https://github.com/vercel/next.js/pull/67673)). Upgrading to newer canary versions of `react` and `react-dom` is not recommended because it will break in production mode.
